guvnor-ala/guvnor-ala-services-api/src/main/java/org/guvnor/ala/services/api/RuntimeQueryResultItem.java

```diff
@@ -0,0 +1,214 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.services.api;
+
+import org.guvnor.ala.services.api.itemlist.PipelineStageItemList;
+
+public class RuntimeQueryResultItem {
+
+    /**
+     * The provider Id of the provider related with this item. Runtimes are always related to a provider so this value
+     * must always be present.
+     */
+    private String providerId;
+
+    private String providerTypeName;
+
+    private String providerVersion;
+
+    /**
+     * When known, it's the id of the pipeline related with this item. It'll be usually the case, but there might be
+     * cases where the runtime was created directly by using the RuntimeProvisioningService and not by a pipeline.
+     */
+    private String pipelineId;
+
+    /**
+     * When known, it's the UUID of the pipeline execution related with this item. It'll be usually the case, but there
+     * might be cases where the runtime was created directly by using the RuntimeProvisioningService and not by a pipeline.
+     */
+    private String pipelineExecutionId;
+
+    /**
+     * When known, it's the execution status of the pipeline related with this item. This status can be on of the
+     * following values. SCHEDULED, RUNNING, FINISHED or ERROR.
+     * It'll be usually the case, but there
+     * might be cases where the runtime was created directly by using the RuntimeProvisioningService and not by a pipeline.
+     */
+    private String pipelineStatus;
+
+    /**
+     * Holds the pipeline execution error message when pipelineStatus == ERROR.
+     */
+    private String pipelineError;
+
+    /**
+     * When a pipeline is involved, holds the information related to the stages of current pipeline. See that there might
+     * be cases where the runtime was created directly by using the RuntimeProvisioningService and not by a pipeline.
+     */
+    private PipelineStageItemList pipelineStageItems;
+
+    /**
+     * When known, it's the id of the runtime related with this item. This value might be un-known if the runtime
+     * wasn't yet created, e.g. when the pipeline is still executing, or the pipeline execution finished but with errors
+     * and no runtime has been produced.
+     */
+    private String runtimeId;
+
+    /**
+     * When known, it's the runtime name for the runtime related wih this item. This value might be un-known if no runtime
+     * name was defined when the runtime was created.
+     */
+    private String runtimeName;
+
+    /**
+     * When known, it's the runtime status. This value might be un-known if the runtime
+     * wasn't yet created, e.g. when the pipeline is still executing.
+     * This status can be on of the following values. STARTED, RUNNING or ERROR.
+     */
+    private String runtimeStatus;
```

Comment:
The case of the runtime status is similar as the one above, but by know with the difference that the runtime status is not yet unified and is also not being returned yet. The runtime status will be part of next sprint.